x/metadata/keeper/query_server.go

```diff
@@ -649,6 +649,35 @@ func (k Keeper) ScopeSpecification(c context.Context, req *types.ScopeSpecificat
 		retval.ScopeSpecification = types.WrapScopeSpecNotFound(specAddr)
 	}
 
+	if found && req.IncludeContractSpecs {
+		for _, id := range spec.ContractSpecIds {
+			cs, ok := k.GetContractSpecification(ctx, id)
+			if !ok {
+				return &retval, fmt.Errorf("error getting scope spec [%s] contract spec [%s]: %w", spec.SpecificationId, id, err)
+			}
+			retval.ContractSpecs = append(retval.ContractSpecs, types.WrapContractSpec(&cs))
+		}
+	}
+
+	if found && req.IncludeRecordSpecs {
+		var err error
+		for _, id := range spec.ContractSpecIds {
+			err = k.IterateRecordSpecsForContractSpec(ctx, id, func(recordSpecID types.MetadataAddress) (stop bool) {
+				rs, ok := k.GetRecordSpecification(ctx, recordSpecID)
+				if !ok {
+					err = fmt.Errorf("error retrieving record spec [%s] for contract spec [%s]", recordSpecID, id)
```

Comment:
I know this is being changed because of the other comment, but I wanted to point something out here.

This `err` value will never be seen. When `IterateRecordSpecsForContractSpec` ends, the `err` value will be set to something returned directly by that function. In this case, it will be `nil` because the iterator didn't encounter any errors.